tests/packages/test_dependency.py

```diff
@@ -294,7 +294,7 @@ def test_with_constraint():
 def test_marker_properly_sets_python_constraint():
     dependency = Dependency("foo", "^1.2.3")
 
-    dependency.marker = 'python_version >= "3.6" and python_version < "4.0"'
+    dependency.marker = 'python_version >= "3.6" and python_version < "4.0"'  # type: ignore[assignment]
```

Comment:
this is an issue with `marker`'s setter accepting the union `str | BaseMarker` and mypy not being able to understand: https://github.com/python/mypy/issues/3004